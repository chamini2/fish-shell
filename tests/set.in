function test_function
    set -l i 1
    for v in ONE TWO THREE
        set -f $v $i
        set i (math "$i + 1")
    end
    echo "I say $ONE, $TWO, $THREE"
    if test $ONE = 1
        set -l ONE 0
        echo -n "Inside if ($ONE) "
    end
    echo "and after ($ONE)"
    if test $ONE
        set ONE 0
        echo -n "Inside if without setting as local ($ONE) "
    end
    echo "and after ($ONE)"
end

function tttt
    set -l local_var 1
    set -f func_var 0
    set -g global_var 2
    for var in func_var local_var global_var
        for flag in local function global
            set -q --$flag $var
            and echo "$var $flag scoped"
        end
    end
    set -eg global_var
    set -ef func_var
    set -el local_var
end

test_function
echo
tttt
