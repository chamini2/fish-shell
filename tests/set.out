I say 1, 2, 3
Inside if (0) and after (1)
Inside if without setting as local (0) and after (0)

func_var local scoped
func_var function scoped
local_var local scoped
local_var function scoped
global_var global scoped
